app/jobs/badge_achievements/send_email_notification_job.rb

```diff
@@ -0,0 +1,10 @@
+module BadgeAchievements
+  class SendEmailNotificationJob < ApplicationJob
+    queue_as :badge_achievements_send_email_notification
+
+    def perform(badge_achievement_id)
+      badge_achievement = BadgeAchievement.find_by(id: badge_achievement_id)
```

Comment:
Badge achievement could be nil because the ID doesn't exist or has been deleted by the time the job runs so remember to exit from the function without sending the email in that case

Reply:
@rhymes solved.